### PR TITLE
build: add "build-container" target

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:

Adds `make build-container` to simplify running the command in releases/README.Dockerfile.md and does some minor cleanup of the Dockerfile.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:
Yes, I know this makes the **image**, not the **container**. But `make build-image` sounded too ambiguous, and I didn't want to mention Docker specifically. Suggestions welcome.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -109,6 +109,13 @@ checksum:
 		shasum -a 256 "$${f}"  | awk '{print $$1}' > "$${f}.sha256" ; \
 	done
 
+.PHONY: build-container
+build-container:
+	docker build --no-cache --build-arg BUILD_DATE=`date -u +"%Y-%m-%dT%H:%M:%SZ"` \
+		--build-arg AKSENGINE_VERSION="$(VERSION)" -t microsoft/aks-engine:$(VERSION) \
```
Comment (tariq1890) [inline Makefile]:
```suggestion
		--build-arg AKSENGINE_VERSION="$(VERSION)" -t microsoft/aks-engine:${VERSION} \
```

File: releases/Dockerfile.linux [hunk 2]
```diff
@@ -1,6 +1,6 @@
 FROM alpine:3.7
```
Comment (tariq1890) [inline releases/Dockerfile.linux]: Can we update this to 3.9?

File: releases/Dockerfile.linux [hunk 3]
```diff
@@ -10,14 +10,14 @@ LABEL maintainer="Microsoft" \
       org.label-schema.name="Azure Kubernetes Engine (aks-engine)" \
       org.label-schema.version=$AKSENGINE_VERSION \
       org.label-schema.license="MIT" \
-      org.label-schema.description="The Azure Kubernetes Engine (aks-engine) generates ARM (Azure Resource Manager) templates for Kubernetes clusters on Microsoft Azure with your choice of DCOS, Kubernetes, or Swarm orchestrators." \
+      org.label-schema.description="The Azure Kubernetes Engine (aks-engine) generates ARM (Azure Resource Manager) templates for Kubernetes clusters on Microsoft Azure." \
```
Comment (tariq1890) [inline releases/Dockerfile.linux]: Thoughts on adding a label for maintainer email?

File: releases/Dockerfile.linux (hunk 2 above)
Comment (mboersma) [reply releases/Dockerfile.linux]: 👍 Done.

File: releases/Dockerfile.linux (hunk 3 above)
Comment (mboersma) [reply releases/Dockerfile.linux]: We could... Maybe change the [maintainer label](https://docs.docker.com/engine/reference/builder/#maintainer-deprecated) from "Microsoft" to an email address if we want to do that. I didn't see anything standardized in [label-schema](http://label-schema.org/rc1/) for maintainer email.

Comment (tariq1890) [reply releases/Dockerfile.linux]: Sure. For later then!

**Pull request merged**